contribute/dev_login_scripts.rst

```diff
@@ -0,0 +1,176 @@
+
+   echo $SHELL
+
+For bash users
+--------------
+
+To invoke it add the following commands to your ~/.bashrc or ~/.profile script.
+
+::
+
+   export OPENCMISS_ROOT=<path-to-where-you-put-your-OpenCMISS-files>
+   if [ -r "$OPENCMISS_ROOT/utilities/scripts/opencmiss_developer.sh" ]; then
+       . $OPENCMISS_ROOT/utilities/scripts/opencmiss_developer.sh
+       fi
+
+For csh/tcsh users
+------------------
+
+To invoke it add the following commands to your ~/.cshrc script.
+
+::
+
+   setenv OPENCMISS_ROOT <path-to-where-you-put-your-OpenCMISS-files>
+   if ( -r "${OPENCMISS_ROOT}/utilities/scripts/opencmiss_developer.csh" ) then
+       source ${OPENCMISS_ROOT}/utilities/scripts/opencmiss_developer.csh
+   endif
+
+Login Script Configuration Options
+==================================
+
+There are a number of options that can be use to configure the OpenCMISS developers login script and to customise its actions. To customise any actions of the login script simply set the variables appropriately before the login script is invoked e.g. For bash users
+
+::
+
+   export OPENCMISS_ROOT=<path-to-where-you-put-your-OpenCMISS-files>
+   export OPENCMISS_TOOLCHAIN=gnu
+   export OPENCMISS_MPI=mpich
+   export OPENCMISS_BUILD_TYPE=debug
+   export OPENCMISS_SETUP_TOTALVIEW=false
+   if [ -r "$OPENCMISS_ROOT/utilities/scripts/opencmiss_developer.sh" ]; then
+       . $OPENCMISS_ROOT/utilities/scripts/opencmiss_developer.sh
+   fi
+
+and for csh/tcsh users
+
+::
+
+   setenv OPENCMISS_ROOT <path-to-where-you-put-your-OpenCMISS-files>
+   setenv OPENCMISS_TOOLCHAIN gnu
+   setenv OPENCMISS_MPI mpich
+   setenv OPENCMISS_BUILD_TYPE debug
+   setenv OPENCMISS_SETUP_TOTALVIEW false
+   if ( -r "${OPENCMISS_ROOT}/utilities/scripts/opencmiss_developer.csh" ) then
+       source ${OPENCMISS_ROOT}/utilities/scripts/opencmiss_developer.csh
+   endif
+
+Details on the individual environment variables are listed below.
+
+.. toctree::
+   :maxdepth: 2
+
+OPENCMISS_ROOT
+--------------
+
+This is the enviroment variable that gives directory path to the OpenCMISS files. It must be set. 
+
+OPENCMISS_INSTALL_ROOT
+----------------------
+
+This environment variable points to where the OpenCMISS files are installed. Optional. If it is not specified it defaults to OPENCMISS_ROOT/install.
+
+OPENCMISS_TOOLCHAIN
+-------------------
+
+This environment variable specifies the toolchain for OpenCMISS to be setup. Available options are gnu and intel. 
+
+OPENCMISS_BUILD_TYPE
+--------------------
+
+This environment variable specifies the build type of OpenCMISS to be setup. Available options are debug, release, relwithdebinfo and minsizerel. Defaults to release.
+
+OPENCMISS_MPI
+-------------
+
+This environment variable specifies the MPI library to use with OpenCMISS. Available options are none, mpich, mpich2, openmpi, mvapich2 or intel. Defaults to none.
```

Comment:
msmpi

Reply:
Not really a msmpi for unix login scripts? i.e., bash/csh/tcsh? Possible with cygwin?

Reply:
Yes I know but I just wanted to cover someone using git bash on Windows, just in case I don't think it likely.  Plus a bit of extra knowledge never hurt.